instrumentation/go.mongodb.org/mongo-driver/config.go

```diff
@@ -25,30 +25,31 @@ const (
 
 // Config is used to configure the mongo tracer.
 type Config struct {
+	TracerProvider trace.Provider
+
 	Tracer trace.Tracer
```

Comment:
Can we make this field unexported? I think following the principle of least privilege here makes sense.

Reply:
Since we are talking about make fields unexported, maybe we could unexport the `Config` itself?

But, after making `Config` unexported, I haven't figured out whether we should keep fields of `Config` exported or not.

Reply:
Yeah, I didn't look too deep, but if there is no need to export the `Config` I'm in favor of making it unexported. :+1:

Reply:
Agreed.  The [style guidelines](https://github.com/open-telemetry/opentelemetry-go/blob/master/CONTRIBUTING.md#config) from the main repo say we should keep the config type unexported unless we expect the user to need to build custom options.  I think it's easier to go from unexported->exported if necessary, so let's try to keep everything unexported until a clear need arises.